solutions/src/main/java/ru/sberbank/school/task02/FxClient.java

```diff
@@ -0,0 +1,81 @@
+package ru.sberbank.school.task02;
+
+import lombok.NonNull;
+import ru.sberbank.school.task02.util.*;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FxClient implements FxClientController {
+
+    private FxConversionService currentConverter;
+
+    FxClient(FxConversionService converter) {
+        currentConverter = converter;
+    }
+
+    @Override
+    public List<FxResponse> fetchResult(@NonNull List<FxRequest> requests) {
+        List<FxResponse> responses = new ArrayList<>();
+        for (FxRequest req : requests) {
+            responses.add(fetchResult(req));
+        }
+        return responses;
+    }
+
+    @Override
+    public FxResponse fetchResult(@NonNull FxRequest requests)
+            throws IllegalArgumentException, NullPointerException {
+        Symbol symbol = getSymbol(requests);
+        BigDecimal amount = getAmount(requests);
+        ClientOperation operation = getOperation(requests);
+
+
+        LocalDateTime localDate = LocalDateTime.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss");
+        String date = localDate.format(formatter);
+
+        BigDecimal price = currentConverter.convert(operation, symbol, amount);
+        return new FxResponse(symbol.getSymbol(),
+                price.toString(),
+                amount.setScale(2, BigDecimal.ROUND_DOWN).toString(),
+                date,
+                operation.toString(),
+                false);
+
+    }
+
+    private BigDecimal getAmount(FxRequest r) throws IllegalArgumentException{
+        BigDecimal amount;
+        try{
+            amount = new BigDecimal(r.getAmount());
+        }catch (IllegalArgumentException e){
+            throw new IllegalArgumentException("Amount is not a number");
+        }
+        return amount;
+    }
+
+    private ClientOperation getOperation(FxRequest r) throws IllegalArgumentException{
+        ClientOperation operation;
+        try{
+            operation = ClientOperation.valueOf(r.getDirection().toUpperCase());
+        }catch (IllegalArgumentException e){
```

Comment:
WhitespaceAround: &#39;catch&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1573365183>*